File: MeetingMinutesFrontend/PFPedro/src/app/services/temas.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Temas } from '../clases/temas';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class TemasService {

  private baseUrl = 'https://mm-project.herokuapp.com/temas/';

  constructor(private http: HttpClient) { }

  /** Métodop para recoger los temas por código de Reunion desde el backend */

  getTemasByCodReunion(codreunion: number): Observable<any> {
    return this.http.get<Temas>(`${this.baseUrl}` + 'tema/' + codreunion);
  }

  /** Método para recoger todos los temas por Serie de reunion desde el backend */

  getAllTemasByCodSReunion(codsreunion: number): Observable<any> {
    return this.http.get<Temas>(`${this.baseUrl}` + 'alltemas/' + codsreunion);
  }

  /** Método para crear un nuevo tema */

  crearTemas(tema: Temas, codreunion: number, codsreunion: number): Observable<any> {
    return this.http.post(`${this.baseUrl}` + 'creartema/' + codreunion + '/' + codsreunion, tema);
  }

  /** Método para añadir información al tema */

  añadirInfoTema(tema: Temas, codTema: number): Observable<any> {
    return this.http.put(`${this.baseUrl}` + 'addinfo/' + codTema, tema);
  }

  /** Método para añadir una decisión al tema */

  añadirDecisionTemas(tema: Temas, codTema: number): Observable<any> {
    return this.http.put(`${this.baseUrl}` + 'adddecision/' + codTema, tema);
  }

  /** Método para cerrar un tema */

  cerrarTemas(tema: Temas): Observable<any> {
    return this.http.put(`${this.baseUrl}` + 'cerrartema/', tema);
  }

  /** Método para recoger los temas de reuniones antiguas a la actual */

  getTemasByCodReunionAntiguaAndNoCerrado(codreunion: number, codsreunion: number): Observable<any> {
    return this.http.get<Temas>(`${this.baseUrl}` + 'temasreuantnocerr/' + codreunion + '/' + codsreunion);
  }

  /** Método para añadir el tema antiguo a la reunión en la que estamos actualmente */

  saveTemaAntiguo(tema: Temas[], codreunion: number): Observable<any> {
    return this.http.post(`${this.baseUrl}` + 'savetemasant/' + codreunion, tema);
  }

  /** Método para recoger un tema por su código desde el backend */

  getTemaByCodTema(codtema: number): Observable<Temas> {
    return this.http.get<Temas>(`${this.baseUrl}` + 'gettema/' + codtema);
  }

  /** Método para modificar un tema */

  modificarTema(tema: Temas): Observable<any> {
    return this.http.put(`${this.baseUrl}` + 'modificarTema/', tema);
  }

  /** Método para añadir un seguimiento a un tema cerrado */

  addSeguimientoTemaCerrado(tema: Temas): Observable<any> {
    return this.http.put(`${this.baseUrl}` + 'addseguimiento', tema);
  }

}

File: MeetingMinutesBackend/MeetingMinutes/src/main/java/com/pedro/service/TemasService.java
/**
 * 
 */
package com.pedro.service;

import java.util.List;

import com.pedro.modelo.Temas;

/**
 * The Interface TemasService.
 *
 * @author Westermeyer
 */
public interface TemasService {
	
	/**
	 * Gets the temas by cod reunion.
	 *
	 * @param codreunion the codreunion
	 * @return the temas by cod reunion
	 */
	List<Temas> getTemasByCodReunion(int codreunion);
	
	/**
	 * Gets the all temas by cod S reunion.
	 *
	 * @param codsreunion the codsreunion
	 * @return the all temas by cod S reunion
	 */
	List<Temas> getAllTemasByCodSReunion(int codsreunion);
	
	/**
	 * Crear temas.
	 *
	 * @param tema the tema
	 * @param codreunion the codreunion
	 * @param codsreunion the codsreunion
	 */
	void crearTemas(Temas tema, int codreunion, int codsreunion);
	
	/**
	 * Añadir info tema.
	 *
	 * @param tema the tema
	 * @param codTema the cod tema
	 */
	void añadirInfoTema(Temas tema, int codTema);
	
	/**
	 * Añadir decision tema.
	 *
	 * @param tema the tema
	 * @param codTema the cod tema
	 */
	void añadirDecisionTema(Temas tema, int codTema);
	
	/**
	 * Cerrar temas.
	 *
	 * @param tema the tema
	 */
	void cerrarTemas(Temas tema);
	
	/**
	 * Gets the temas by cod reunion antigua and no cerrado.
	 *
	 * @param codreunion the codreunion
	 * @param codsreunion the codsreunion
	 * @return the temas by cod reunion antigua and no cerrado
	 */
	List<Temas> getTemasByCodReunionAntiguaAndNoCerrado(int codreunion, int codsreunion);
	
	/**
	 * Save tema antiguo.
	 *
	 * @param tema the tema
	 * @param codreunion the codreunion
	 */
	void saveTemaAntiguo(Temas[] tema, int codreunion);
	
	/**
	 * Gets the tema by cod tema.
	 *
	 * @param codtema the codtema
	 * @return the tema by cod tema
	 */
	Temas getTemaByCodTema(int codtema);
	
	/**
	 * Modificar tema.
	 *
	 * @param tema the tema
	 */
	void modificarTema(Temas tema);
	
	/**
	 * Añadir seguimiento tema cerrado.
	 *
	 * @param tema the tema
	 */
	void añadirSeguimientoTemaCerrado(Temas tema);
	
}
